spec/controllers/shipping_api_controller_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'rails_helper'
+
+RSpec.describe ShippingApiController, type: :controller do
+
+  describe "POST rates" do
+    let(:params) {
+      { origin: { zip: "98101", city: "Seattle", state: "WA", country: "US"},
+        destination: { city: "Beaverton", state: "OR", zip: 97005, country: "USA" },
+        package: { weight: 12, dimensions: [15, 10, 4.5], units: :imperial }
+      }
+    }
+
+
+    context "when params are valid" do
+      before :each do
+        VCR.use_cassette("/rates", record: :new_episodes) do
+          post :rates, params
+        end
+      end
+
+      it "is successful" do
+        expect(response.response_code).to eq(200)
+      end
+
+      it "returns json" do
+        expect(response.header['Content-Type']).to include 'application/json'
+      end
+
+      describe "the returned json object" do
+        before :each do
+          VCR.use_cassette("/rates", record: :new_episodes) do
+            post :rates, params
```

Comment:
Doesn't this end up making the request twice? Once at line 17 and the second time here?